Sources/_OpenAPIGeneratorCore/Translator/Responses/translateResponse.swift

```diff
@@ -103,6 +103,61 @@ extension TypesFileTranslator {
                     )
                 )
                 bodyCases.append(bodyCase)
+
+                var throwingGetterSwitchCases = [
+                    SwitchCaseDescription(
+                        kind: .case(.identifier(".\(identifier)"), ["body"]),
+                        body: [.expression(.return(.identifier("body")))]
+                    )
+                ]
+                // We only generate the default branch if there is more than one case to prevent
+                // a warning when compiling the generated code.
+                if typedContents.count > 1 {
+                    throwingGetterSwitchCases.append(
+                        SwitchCaseDescription(
+                            kind: .default,
+                            body: [
+                                .expression(
+                                    .try(
+                                        .identifier("throwUnexpectedResponseBody")
+                                            .call([
+                                                .init(
+                                                    label: "expectedContent",
+                                                    expression: .literal(.string(contentType.headerValueForSending))
```

Comment:
```suggestion
                                                    expression: .literal(.string(contentType.headerValueForValidation))
```

While this is just for printing, `headerValueForValidation` is more appropriate as it doesn't include `charset`, which isn't used to match content types.

Reply:
After some further discussion, the right thing to do is probably use the `headerValueForValidation` here, because that's actually what was used to deserialize the body.

However, the fact that `headerValueForValidation` here does _not_ include the additional parameter, means we've probably uncovered an existing bug with how we handle responses when there are overlapping content types. We should deserialize as the most specific one, but we're probably at the mercy of the ordering of the deserialization logic, which, in turn, is probably derived from the order in which the contents appear in the OpenAPI document.

Reply:
Updated in 0905ce0.

Reply:
Thanks. Please file an issue when you have a moment for improving the matching when content types overlap.

Reply:
Done: https://github.com/apple/swift-openapi-generator/issues/315